src/bsp/src/stm32/src/BSP.cpp

```diff
@@ -1,16 +1,29 @@
 #include "BSP.h"
-#include "bsp/SettingsContainer.h"
+#include "UserInterface.h"
+#include "bsp/BSPContainer.h"
 #include <hal/hal.h>
 #include <logger/LoggerContainer.h>
 
 BSP::BSP() : m_storage(LoggerContainer::getLogger()) {}
 BSP::~BSP() = default;
 
+// Example button callback that cycles through the RGB possibilities
+void BSP::buttonCallback(void* context) {
+    static_cast<BSP*>(context)->m_currentRGBState += 1;
+    static_cast<BSP*>(context)->m_currentRGBState %= 8;
+    dynamic_cast<UserInterface&>(BSPContainer::getUserInterface())
+        .setRGBLed(static_cast<RgbColor>(static_cast<BSP*>(context)->m_currentRGBState));
+}
+
 void BSP::initChip(void* args) {
     (void)args;
 
     Hal_init();
     m_storage.loadFromFlash();
+    dynamic_cast<UserInterface&>(BSPContainer::getUserInterface())
+        .setButtonCallback(Button::BUTTON_0, buttonCallback, this);
+    dynamic_cast<UserInterface&>(BSPContainer::getUserInterface())
```

Comment:
Dynamic casts needs the RTTI. What is the overhead of the RTTI? Should we compile with -fno-rtti (for embedded) and just use c style or reinterpret cast when necessary? Valid for all dynamic cast calls

Reply:
Yeah good idead. I added -fno-rtti for C++ targets and changed for a reintrepret_cast. Anyways, this code will not be permanent